File: C++/2491. Divide Players Into Teams of Equal Skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        int n = skill.size();
        sort(skill.begin(), skill.end());
        long long sum = 0;
        long long tempsum = skill[0] + skill[n - 1];
        for (int i = 0, j = n - 1; i < j; ++i, --j) {
            long long currsum = skill[i] + skill[j];
            if (tempsum != currsum) {
                return -1;
            }
            sum += (long long)skill[i] * skill[j];
        }
        return sum;
    }
};

File: JAVA/2491. Divide Players Into Teams of Equal Skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);       
        long product = 0;
        int target = skill[0] + skill[skill.length - 1];
        int i = 0;
        int j = skill.length - 1;
        while (i < j) {
            if (skill[i] + skill[j] != target) return -1;
            product = product + (skill[i] * skill[j]);
            i++;
            j--;
        }
        return product;
    }
}
